src/docs/output-targets/custom-elements.md

```diff
@@ -37,24 +37,34 @@ import { HelloWorld } from 'my-library/dist/components/hello-world';
-your components and their respective `defineCustomElement` helper functions,
-looking something like this:
+The output directory will also contain an `index.js` file which exports some helper methods by default. The contents of the file
+will look something like (the contents may look different if [`customElementsExportBehavior`](#customelementsexportbehavior) is specified):
 
-```tsx
+```js
 export { setAssetPath, setPlatformOptions } from '@stencil/core/internal/client';
-export { MyComponent, defineCustomElement as defineCustomElementMyComponent } from './my-component.js';
-export {
-  MyOtherComponent,
-  defineCustomElement as defineCustomElementMyOtherComponent
-} from './my-other-component.js';
 ```
 
-This file can be used as the root module when distributing your component
-library, see [below](#distributing-custom-elements) for more details.
-
 ## Config
 
+### customElementsExportBehavior
+
+This config option provides additional behaviors that will alter the default component export OR custom element definition behaviors
+for this target. The desired behavior can be set via the following in a project's Stencil config:
+
+```ts
+{
+  type: 'dist-custom-elements',
+  customElementsExportBehavior: 'default' | 'single-export-module'
+}
+```
+
+| Option                 | Description                                                                                                                                                                                                                                                                                                                                                                                               |
+| ---------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `default`              | No additional re-export or auto-definition behavior will be performed.<br><br>This value will be used if no explicit value is set in the config, or if a given value is not a valid option.                                                                                                                                                                                                               |
+| `single-export-module` | All component and custom element definition helper functions will be exported from the `index.js` file in the output directory (see [Defining Exported Custom Elements](#defining-exported-custom-elements) for more information on this file's purpose). This file can be used as the root module when distributing your component library, see [below](#distributing-custom-elements) for more details. |
```

Comment:
Should we add a comment about treeshaking here and that it won't always be possible to fully treeshake with this option?

Reply:
It should always be possible once `autoDefineCustomElements` is migrated to be one of these options and not its own flag. I could add a comment for now with a "todo" to remove it later, but all that should happen before the v3 release anyway 🤷‍♂️